packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,254 @@
+// @ts-check
+
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { fit, M, makeCopyBag, makeStore } from '@agoric/store';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+
+const { details: X } = assert;
+
+export const KW = /** @type { const } */ ({
+  /** seat keyword for use in offers to return an attestation. */
+  Attestation: 'Attestation',
+});
+
+/**
+ * Find-or-create value in store.
+ *
+ * @type {<K, V>(store: Store<K, V>, key: K, thunk: () => V) => V}
+ */
+const getOrElse = (store, key, make) => {
+  if (store.has(key)) {
+    return store.get(key);
+  }
+  const value = make();
+  store.init(key, value);
+  return value;
+};
+
+/**
+ * To support `makeAttestation`, we directly mint attestation payments.
+ * We still use a ZCFMint because returning attestations is done
+ * through Zoe offers.
+ *
+ * @param {ZCF} zcf
+ * @param {ZCFMint} zcfMint
+ * @param {Amount} amountToMint
+ * @returns {Promise<Payment>}
+ */
+const mintZCFMintPayment = (zcf, zcfMint, amountToMint) => {
+  const { userSeat, zcfSeat } = zcf.makeEmptySeatKit();
+  zcfMint.mintGains(harden({ [KW.Attestation]: amountToMint }), zcfSeat);
+  zcfSeat.exit();
+  return E(userSeat).getPayout(KW.Attestation);
+};
+
+/**
+ * Make an ZCFMint attentuated so that minting represents
+ * putting a lien on staked assets. Payments can be returned,
+ * releasing (part of) the lien.
+ *
+ * NOTE: Using Issuer.combine or AmountMath.add could result in
+ * Payments / Amounts with multiple addresses.
+ * Callers are responsible to split any such payment before returning.
+ *
+ * @param {ZCF} zcf
+ * @param {Brand<'nat'>} stakeBrand brand of the staked assets
+ * @param {ERef<StakingAuthority>} lienBridge bridge to account state
+ */
+const makeAttestationKit = async (zcf, stakeBrand, lienBridge) => {
+  const { add, subtract, makeEmpty, isGTE, coerce } = AmountMath;
+  const empty = makeEmpty(stakeBrand);
+  const zcfMint = await zcf.makeZCFMint(KW.Attestation, AssetKind.COPY_BAG);
```

Comment:
I'd prefer this was `attMint` or `lienMint` or even `nftMint`.  When I see `zcfMint` I think of a default fungible mint.